src/Symfony/Component/Security/Http/Authenticator/QueryAccessTokenAuthenticator.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\Authenticator;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Security\Core\User\UserProviderInterface;
+use Symfony\Component\Security\Http\Authentication\AuthenticationFailureHandlerInterface;
+use Symfony\Component\Security\Http\Authentication\AuthenticationSuccessHandlerInterface;
+
+/**
+ * Provides an implementation of the RFC6750 of an authentication via
+ * an access token in the Query String.
+ *
+ * @author Florent Morselli <florent.morselli@spomky-labs.com>
+ *
+ * @final
+ */
+class QueryAccessTokenAuthenticator extends AbstractAccessTokenAuthenticator
```

Comment:
Even if it's in the RFC, I don't recommend to support query parameter:
> Because of the security weaknesses associated with the URI method
   (see [Section 5](https://datatracker.ietf.org/doc/html/rfc6750#section-5)), including the high likelihood that the URL
   containing the access token will be logged, it SHOULD NOT be used
   unless it is impossible to transport the access token in the
   "Authorization" request header field or the HTTP request entity-body.
   Resource servers MAY support this method.

Reply:
I agree. The query string method should not be recommended at all. I mentioned the same section in the documentation.
Same goes for the request body one. There are limitations that show it is not usable at all.